docker-compose.yml

```
@@ -1,100 +1,258 @@
 version: "3.9"
-# Please refer the documentation located at https://docs.mathesar.org/installation/docker-compose/ for the setting up Mathesar using this docker-compose.yml.
+#-------------------------------
+# Mathesar's docker compose file
+#-------------------------------
+
+
+#-------------------------------------------------------
+# PREREQUISITES:
+#-------------------------------------------------------
+# OS: Linux, Mac, Windows(WSL).
+# Docker v23+ ($: sudo docker version)
+# Docker Compose v2.10+ ($: sudo docker compose version)
+#-------------------------------------------------------
+
+
+#--------------------------------------------------
+# CONFIG:
+#--------------------------------------------------
+x-config: &config
+  # (Required) Replace '?' with '-' followed by your 50 character long Django
+  # SECRET_KEY, you can generate one here: https://djecrety.ir/
+  SECRET_KEY: ${SECRET_KEY:?}
```

I'm debating whether we should include a default key that's obviously unfit for production. Something like:


```suggestion
  SECRET_KEY: ${SECRET_KEY:-PLEASEDONOTUSETHISSECRETKEYITISSUPERDUPERINSECURE}
```

---

After some discussion in various channels, I think the current failure state is better in this use-case.